docs/migration.md

```diff
@@ -0,0 +1,38 @@
+Migration is pretty easy. Remember you can always refer to the [updated demo](https://github.com/faceyspacey/redux-first-router-demo)
+
+In v1, `history` was a peerDependency. We have built our own history management tool, its actually part of [Rudy](https://github.com/respond-framework/rudy), but we have implemented it into RFR2. 
+
+While RFR2 is a breaking change, its trivial to get up and running again. 
+
+Inside of your `configureStore.js` [file](https://github.com/scriptedalchemy/redux-first-router-demo/blob/6c8238eee713ce0079aeae1ce328d305bddd0ee3/src/configureStore.js#L11),
+
+This package requires ***Node 8 and up***
+
+Change this:
+```js
+  const { reducer, middleware, enhancer, thunk } = connectRoutes(
+    history,
+    routesMap,
+    options
+  )
+```
+To this:
+
+```js
+  const { reducer, middleware, enhancer, thunk } = connectRoutes(
+    routesMap, {
+    ...options,
+    initialEntries
```

Comment:
should be `...initialEntries`, right?

Reply:
no, initialEntries is the key

Reply:
```js
/* eslint-env browser */

import { createStore, applyMiddleware, compose, combineReducers } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension/logOnlyInProduction'
import {
  push,
  replace,
  jump,
  back,
  next,
  reset,
  set,
  setParams,
  setQuery,
  setState,
  setHash,
  setBasename,
  createRouter,
} from '@respond-framework/rudy'

import routes from './routes'
import * as reducers from './reducers'

export default (preloadedState, initialEntries) => {
  const options = { initialEntries, basenames: ['/foo', '/bar'] }
  const { reducer, middleware, firstRoute, history, ctx } = createRouter(
    routes,
    options,
  )
  const rootReducer = combineReducers({ ...reducers, location: reducer })
  const middlewares = applyMiddleware(middleware)
  const enhancers = composeEnhancers(middlewares)
  const store = createStore(rootReducer, preloadedState, enhancers)

  if (module.hot) {
    module.hot.accept('./reducers/index', () => {
      const newRootReducer = combineReducers({ ...reducers, location: reducer })
      store.replaceReducer(newRootReducer)
    })
  }

  if (typeof window !== 'undefined') {
    window.routes = routes
    window.store = store
    window.hist = history
    window.actions = actionCreators
    window.ctx = ctx
  } else {
    global.routes = routes
    global.store = store
    global.hist = history
    global.actions = actionCreators
    global.ctx = ctx
  }

  return { store, firstRoute }
}

const composeEnhancers = (...args) =>
  typeof window !== 'undefined'
    ? composeWithDevTools({ actionCreators })(...args)
    : compose(...args)

const actionCreators = {
  push,
  replace,
  jump,
  back,
  next,
  reset,
  set,
  setParams,
  setQuery,
  setState,
  setHash,
  setBasename,
}
```

Reply:
Something is wrong to me.
In this part of the code https://github.com/faceyspacey/redux-first-router-demo/blob/rudy/src/index.js
There is some commented code. The second parameter is an object. So we need to destructurate it, right?
Or we want the key to be related to an object?
I use destructuration in my code right now and it works well. Maybe because my extra parameter is not being cared.

Reply:
oh shoot, sorry this is Rudy 

import { createStore, applyMiddleware, compose, combineReducers } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension/logOnlyInProduction'
import { connectRoutes } from 'redux-first-router'

import routesMap from './routesMap'
import options from './options'
import * as reducers from './reducers'
import * as actionCreators from './actions'

export default (preLoadedState, initialEntries) => {
  const { reducer, middleware, enhancer, thunk } = connectRoutes(routesMap, {
    ...options,
    initialEntries
  })

  const rootReducer = combineReducers({ ...reducers, location: reducer })
  const middlewares = applyMiddleware(middleware)
  const enhancers = composeEnhancers(enhancer, middlewares)
  const store = createStore(rootReducer, preLoadedState, enhancers)

  if (module.hot && process.env.NODE_ENV === 'development') {
    module.hot.accept('./reducers/index', () => {
      const reducers = require('./reducers/index')
      const rootReducer = combineReducers({ ...reducers, location: reducer })
      store.replaceReducer(rootReducer)
    })
  }

  return { store, thunk }
}

const composeEnhancers = (...args) =>
  typeof window !== 'undefined'
    ? composeWithDevTools({ actionCreators })(...args)
    : compose(...args)